piptools/writer.py

```diff
@@ -32,7 +32,8 @@
 
 MESSAGE_UNSAFE_PACKAGES_UNPINNED = comment(
     "# WARNING: The following packages were not pinned, but pip requires them to be"
-    "\n# pinned when the requirements file includes hashes. "
+    "\n# pinned when the requirements file includes hashes and the requirement is not "
```

Comment:
Let's not print out a trailing space — the line continuation adds a newline right away.
```suggestion
    "\n# pinned when the requirements file includes hashes and the requirement is not"
```